rest/utilities_testing.go

```diff
@@ -231,7 +230,39 @@ func (rt *RestTester) Bucket() base.Bucket {
 
 		rt.DatabaseConfig.SGReplicateEnabled = base.BoolPtr(rt.RestTesterConfig.sgReplicateEnabled)
 
-		_, err = rt.RestTesterServerContext.AddDatabaseFromConfig(*rt.DatabaseConfig)
+		if rt.RestTesterConfig.TestBucket != nil {
+			// If using a leaky bucket, make sure it ignores attempted closures to avoid double closures panic
+			// due to closing both RestTesterServerContext and testBucket in rt.Close()
+			leakyBucket, isLeaky := base.AsLeakyBucket(testBucket)
+			if isLeaky {
+				leakyBucket.SetIgnoreClose(true)
+			}
```

Comment:
Reduce scope of this to avoid accidental usage of `leakyBucket` even if `isLeaky` is false

(this suggestion makes `leakyBucket` only accessible inside the if block)

```suggestion
			if leakyBucket, isLeaky := base.AsLeakyBucket(testBucket); isLeaky {
				leakyBucket.SetIgnoreClose(true)
			}
```